### ISSUE TITLE
Coinbase transactions are only indexed if their outputs have been spent as inputs of a new tx

### ISSUE THREAD

**Issue by cronicc** [HorizenOfficial/zen#352]
If an address has only ever received coinbase transactions, it is not indexed by the block explorer.

Early testnet blocks are a good example of this:
https://explorer-testnet.zensystem.io/block/00143a02152cf0cf3b6cbb3d211f51d748c5dbd433c2033bbcafd60402582874 - 12.5 zen going to address ztWuqiGBEHkKSMBgbrbzTZbHudPzY2jE4x8
https://explorer-testnet.zensystem.io/address/ztWuqiGBEHkKSMBgbrbzTZbHudPzY2jE4x8 - address not showing a balance

```
zen-cli --testnet getaddresstxids '{"addresses": ["ztWuqiGBEHkKSMBgbrbzTZbHudPzY2jE4x8"]}'
[
]
zen-cli --testnet getaddressbalance '{"addresses": ["ztWuqiGBEHkKSMBgbrbzTZbHudPzY2jE4x8"]}'
{
  "balance": 0,
  "received": 0
}
zen-cli --testnet getaddressdeltas '{"addresses": ["ztWuqiGBEHkKSMBgbrbzTZbHudPzY2jE4x8"]}'
[
]
zen-cli --testnet getaddressutxos '{"addresses": ["ztWuqiGBEHkKSMBgbrbzTZbHudPzY2jE4x8"]}'
[
]
```

If the coinbase tx vout become vin of a new transaction, they get indexed.

The issue likely is in this part of the code https://github.com/HorizenOfficial/zen/blob/3aaeab91ea4608fbb3f92915d0ac588aafc0f2f2/src/main.cpp#L2565-L2578

Only if a tx is `!tx.IsCoinBase()` it gets added to AddressIndex and SpentIndex, we would need an else specific for AddressIndexing here where coinbase transactions are handled.

**Comment by JackPiri** [comment]
The issue is caused by missing support for P2PK transactions (hence no relation with coinbaseness or unspentness). 

In particular, when retrieving the script  type:
https://github.com/HorizenOfficial/zen/blob/3aaeab91ea4608fbb3f92915d0ac588aafc0f2f2/src/main.cpp#L2585-L2594
only P2PKH and P2SH type are supported. This results in the transaction not being indexed at all.

Support for P2PK script type is added in branch "support_indexing_for_P2PK".

**Comment by JackPiri** [comment]
Closed by commit 82846eb9639b4111adb553bebec2e618f5b05188
